### ISSUE TITLE
Moving through stacks (feature request)

### ISSUE THREAD

**Issue by RickDW** [pop-os/shell#655]
I have been enjoying the stacked windows feature that was implemented a while ago, I think it's a great addition to pop shell, however there is an issue that makes navigating through windows a pain. Let's say I have two windows in a stack on the left half of my screen, and that stack is tiled with a single window on the right half of my screen (shown in the screenshot). If I want to keep the left-most window in the stack open when I navigate to the right-most window on the screen, I have no choice but to first navigate to the right-most window of the _stack_ and then go on to the right-most window of the screen. This will hide the window that I wanted to stay visible. 

![StackNavigation](https://user-images.githubusercontent.com/25059442/97086499-de0d1d80-1623-11eb-8df1-9f112a75d6fc.png)

It is possible to do what I want if I would use my mouse to directly focus the right-most window on screen, but for me this defeats the point of using a keyboard driven approach. 

The solution that I think would be good, is to use super + ctrl + left/right (or h/l), as those are currently not assigned to anything in pop shell. (super+ctrl+up/down (j/k) is used for navigating workspaces, making the keyboard combinations slightly confusing perhaps, but I don't think this would be a big issue). Super+ctrl+left/right (or h/l) could then be used to navigate horizontally, but when the currently active window is located in a stack, it will skip all windows in the stack and focus on the first window in the direction indicated by the user. This stack skipping would only have to be implemented in the horizontal direction, as the current functionality allows for vertical navigation without changing the visible window of a stack.

**Comment by arnogon** [comment]
Sway/i3 achieve this with `focus parent`. In your example `<Super>a` will shift focus from the stacked window to the parent container (the whole stack), and all tabs will be highlighted. From there you can use `<Super>l` to shift focus to the window on the right without going through the hidden window in the stack.

`focus parent` mode has other [use cases](https://i3wm.org/docs/userguide.html#_focus_parent) too. It would be great if this feature is added to Pop Shell.

**Comment by RickDW** [comment]
Exactly, I didn't expect it to be possible in pop shell since it sounds like most of the window management would need to be redesigned and implemented, but it would be a much better solution than what I proposed. I'm thinking of letting you move around an entire stack of windows for example, which would definitely be welcome feature. 

If you could only select the stack as a whole and manipulate it with all of the keybindings that pop shell gives us like any other window, we'd be golden. Having stacks within stacks within stacks and so on, like i3 has, would maybe make things more flexible, but I don't think that'd be necessary. ( Might want to ask i3 power users though, it might be more useful than I think. )

**Comment by the1derer** [comment]
This is a much needed feature.

**Comment by SRNissen** [comment]
Turns out this is three related issues:

https://github.com/pop-os/shell/issues/655

https://github.com/pop-os/shell/issues/1639

https://github.com/pop-os/shell/issues/1531
